bridge/contracts/NativeERCBridgePoolBase.sol

```diff
@@ -0,0 +1,137 @@
+// SPDX-License-Identifier: MIT-open-group
+pragma solidity ^0.8.16;
+import "contracts/interfaces/IBridgePool.sol";
+import "contracts/utils/ImmutableAuth.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "contracts/libraries/errors/NativeERCBridgePoolBaseErrors.sol";
+import "contracts/utils/AccusationsLibrary.sol";
+import "contracts/libraries/parsers/MerkleProofParserLibrary.sol";
+import "contracts/libraries/parsers/VSPreImageParserLibrary.sol";
+import "contracts/libraries/parsers/PClaimsParserLibrary.sol";
+import "contracts/utils/MerkleProofLibrary.sol";
+import "contracts/Snapshots.sol";
+
+abstract contract NativeERCBridgePoolBase is ImmutableFactory, ImmutableBridgeRouter, IBridgePool {
+    using MerkleProofParserLibrary for bytes;
+    using MerkleProofLibrary for MerkleProofParserLibrary.MerkleProof;
+
+    struct DepositParameters {
+        uint256 tokenId;
+        uint256 tokenAmount;
+    }
+
+    address private immutable _snapshotsAddress;
+
+    mapping(bytes32 => bool) private _consumedUTXOIDs;
+
+    constructor(address alicenetFactoryAddress, address snapshotsAddress_)
+        ImmutableFactory(alicenetFactoryAddress)
```

Comment:
if you are passing snapshots down from the inheritting contract I don't think you need factoryAddress() at this level

Reply:
Removed ImmutableSnapshots and ImmutableBridgeRouter from base but we still need ImmutableBridgePoolFactory for onlyBridgePoolFactory. Or should we pass BPFactory address as constructor parameter too?